packages/agreement-process/src/routers/AgreementRouter.ts

```diff
@@ -90,11 +92,20 @@ const agreementRouter = (
     "/agreements/:agreementId/submit",
     authorizationMiddleware([ADMIN_ROLE]),
     async (req, res) => {
+      const loggerInstance = logger({
+        correlationId: req.ctx.correlationId,
+        userId: req.ctx.authData.userId,
+        organizationId: req.ctx.authData.organizationId,
+      });
+      const makeApiProblem = makeApiProblemBuilder(loggerInstance, errorCodes);
```

Comment:
Could these 5 lines of code repeated everywhere in the router become a function? Something to be used like:
`const { loggerInstance, makeApiProblem } = loggerAndMakeApiProblemBuilder(req: Request)`